parser-typechecker/src/Unison/Codebase/TranscriptParser.hs

```diff
@@ -150,10 +152,12 @@ run dir configFile stanzas codebase = do
       awaitInput = do
         cmd <- atomically (Q.tryDequeue cmdQueue)
         case cmd of
+          -- end of ucm block
           Just Nothing -> do
-            output "\n```\n" -- this ends the ucm block
-            writeIORef hidden Shown
+            output "\n```\n"
+            dieUnexpectedSuccess
```

Comment:
At the end of the `ucm` block check if there was an unexpected success
There is no need to update the `hidden`, as the next stanza will do that.